unfurl/server.py

```diff
@@ -50,31 +56,97 @@ def health():
 
 
 @app.route("/export")
-@cache.cached(query_string=True) # Ensure that the request cached includes the query string (the response differs between different formats)
+@cache.cached(
+    query_string=True
+)  # Ensure that the request cached includes the query string (the response differs between different formats)
 def export():
     requested_format = request.args.get("format", "deployment")
     if requested_format not in ["blueprint", "environments", "deployment"]:
-        return jsonify({
-            "code": "BAD_REQUEST",
-            "message": "Query parameter 'format' must be one of 'blueprint', 'environments' or 'deployment'"
-        }), 400 # BAD_REQUEST
+        return create_error_response("BAD_REQUEST", "Query parameter 'format' must be one of 'blueprint', 'environments' or 'deployment'")
+        
+
+    # Default to exporting the ensemble provided to the server on startup
+    path = current_app.config["UNFURL_ENSEMBLE_PATH"]
+
+    clone_root = current_app.config["UNFURL_CLONE_ROOT"]
+
+    # If asking for external repository
+    if request.args.get("url") is not None:
+        git_url = unquote(request.args.get("url"))  # Unescape the URL
+        repo = LocalEnv(
+            current_app.config["UNFURL_ENSEMBLE_PATH"], can_be_empty=True
+        ).find_git_repo(git_url)
+
+        # Repo doesn't exists, clone it
+        if not repo:
+            from . import init
+
+            init.clone(
```

Comment:
clone() is kind of lame because its doesn't return success or failure.. for logging let's at least print the message it returns